community/cypher-shell/cypher-shell/src/main/java/org/neo4j/shell/cli/NonInteractiveShellRunner.java

```diff
@@ -80,6 +80,11 @@ public int runUntilEnd()
         }
 
         int exitCode = EXIT_SUCCESS;
+
+        if(statements.size()==0) {
```

Comment:
Note, there are also cases when `statements.size()` is larger than 0 and we hit this bug. No need to change though, I've made an alternative patch already.

Reply:
Thanks